src/watch.c

```diff
@@ -110,9 +110,9 @@ watch_index_handler(struct watch_handler *handler, enum watch_event event, enum
 		return check_file_mtime(&handler->last_modified, "%s/index", repo.git_dir)
 			? check : WATCH_NONE;
 
-	if (!check_file_mtime(&handler->last_modified, "%s/index", repo.git_dir) ||
-	    event == WATCH_EVENT_SWITCH_VIEW ||
-	    !index_diff(&diff, opt_show_untracked, false))
+	if (event == WATCH_EVENT_SWITCH_VIEW ||
+	    !(check_file_mtime(&handler->last_modified, "%s/index", repo.git_dir) ||
+	      index_diff(&diff, opt_show_untracked, false)))
```

Comment:
👍 better captures the boolean logic here.

Reply:
Had to change the !(a || b) to (!a && !b) because of a Valgrind false positive.